GMG/Scenes/ChordProgress/ChordProgressView.swift

```diff
@@ -306,50 +358,44 @@ extension ChordProgressView {
             segmentStartTime + segmentDuration
         }
 
-        private var filteredChordCells: [ChordCell] {
-            var filteredChordCells: [ChordCell] = chordCells.filter {
-                chordCell in
-                segmentStartTime <= chordCell.startTime
-                    && chordCell.startTime < segmentEndTime
-            }
-
-            /// 현재 세그먼트에 코드가 없을 경우 추가
-            if filteredChordCells.isEmpty,
-                let previousChordCell: ChordCell = chordCells.filter({
+        private var targetChordCells: [ChordCell] {
+            var targetChordCells: [ChordCell] = chordCells.filter {
                     chordCell in
-                    chordCell.startTime <= segmentStartTime
-                }).last
-            {
-                filteredChordCells = [previousChordCell]
+                    segmentStartTime <= chordCell.startTime
+                        && chordCell.startTime < segmentEndTime
+                }
+            
+            let previousChordCell: ChordCell = {
+                guard let previous = chordCells.filter({ c in c.startTime <= segmentStartTime }).last
+                else {
+                    return ChordCell(
+                        chord: nil
+                        , chordCandidates: []
+                        , startTime: segmentStartTime
+                    )
+                }
+                
+                return previous
+            }()
```

Comment:
익명 함수로 하신 이유가 있으신가요?

```suggestion
            let previousChordCell: ChordCell =
                chordCells
                .filter { chordCell in
                    chordCell.startTime <= segmentStartTime
                }
                .last
                ?? ChordCell(
                    chord: nil,
                    chordCandidates: [],
                    startTime: segmentStartTime
                )
```

Reply:
previouseChordCell을 구하려면, filter를 수행한 이후에, null 체크를 해야하는 조건이 필요해서 하나의 context로 묶어 익명함수 처리를 했습니다.

근데 filter 이후에 필요한 로직이 null 체크 하나 뿐이라면 레몬이 제안한 방식이 더 좋겠네요~!